editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx

```diff
@@ -0,0 +1,490 @@
+                                value: "National Aquarium Dubai",
+                            },
+                            {
+                                type: "string",
+                                key: "subtitle",
+                                displayer: "subtitle",
+                                value: "Brutalism, Portraits"
+                            }
+                            ,
+                            {
+                                type:"color",
+                                key:"backgroundColor",
+                                displayer:"Background Color",
+                                value:"#f1f1e8"
+                            }
+                        ]
+                    }
+            ]
+        });
+
+        this.addProp({
+            type:"number",
+            key:"cardRepeatTime",
+            displayer:"Repeat Count",
+            value:2
+        });
+
+
+     
+    }
+    
+
+      getName(): string {
+        return "Image Gallery 7";
+      }
+    
+      render() {
+  const gallery = this.getPropValue("gallery") as CardType[];
+  const repeatCount = this.getPropValue("cardRepeatTime") as number;
+  const cardBackgroundColor = this.getPropValue("backgroundColor") as string || "#f1f1e8";
+ 
+  const repeatedGallery = Array.from({ length: repeatCount }).flatMap(() => gallery);
+
+  return (
+    <div
+      className={this.decorateCSS("imageGallery7")}
+      style={{
+        "--composer-card-background": cardBackgroundColor,
+      } as React.CSSProperties}
+    >
+      <div className={this.decorateCSS("content")}>
+        {Array.from({ length: 4 }).map((_, columnIndex: number) => (
+          <div key={columnIndex} className={`${this.decorateCSS("column")} ${this.decorateCSS(`column-${columnIndex}`)}`}>
+            {repeatedGallery
+              .filter((_, index: number) => index % 4 === columnIndex)
+              .map((card, index) => {
+                const image = card.value.find(item => item.type === "image") as ImageType;
+                const title = card.value.find(item => item.key === "title") as unknown as { type: "string"; key: string; displayer: string; value: string };
+                const subtitle = card.value.find(item => item.key === "subtitle") as unknown as  { type: "string"; key: string; displayer: string; value: string };
+                const backgroundColor = card.value.find(item => item.key === "backgroundColor")?.value as string || cardBackgroundColor;
+
+                return (
+                  <div key={index} className={this.decorateCSS("card")} style={{ backgroundColor }}>
+                    <img src={image.value} alt={title.value} className={this.decorateCSS("image")} />
+                    <div className={this.decorateCSS("text-container")}>
+                      <div className={this.decorateCSS("title")}>{title.value}</div>
+                      <div className={this.decorateCSS("subtitle")}>{subtitle.value}</div>
```

Comment:
don't render relevant elements if prop is empty
[see here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)